daft/runners/ray_runner.py

```diff
@@ -731,9 +731,12 @@ def __init__(
         max_task_backlog: int | None,
     ) -> None:
         super().__init__()
+
+        # Reuse the existing Ray context if already initialized
         if ray.is_initialized():
-            logger.warning("Ray has already been initialized, Daft will reuse the existing Ray context.")
-        self.ray_context = ray.init(address=address, ignore_reinit_error=True)
+            self.ray_context = ray.init(ignore_reinit_error=True)
```

Comment:
If I understand this correctly, this would ignore the provided address if ray is already initialized. Would that cause problems if the user is running the code on an existing Ray instance but specified a connection to a different one?

Reply:
Yes correct. In that case we actually raise a warning that the address is ignored, but that warning is raised upstream in `set_runner_ray`. I am moving it here instead so that the warning happens instead in `RayRunner.__init__`